File: Lab/Lab6/UseSizeOf.java

import java.io.PrintWriter;
import com.vladium.utils.IObjectProfileNode;
import com.vladium.utils.ObjectProfileFilters;
import com.vladium.utils.ObjectProfileVisitors;
import com.vladium.utils.ObjectProfiler;

/*
Andreas Landgrebe
March 29, 2014
Laboratory assignment #6
In order to implement this properly, one needs to set their classpath to the jar file that is being asked for the laboratorty assignment
*/

//These are the import statements to properly use the jar file to see the size of the primitive data types that have been declared
public class UseSizeOf{

	public static void main (String [] args){

		short s = 0;
		int i = 0;
		long l = 0;
		float f = 0;
		double d = 0.0;
		boolean b = true;
		char c = 0;
		//This are all of the data types that are being used to see the size in bytes and bits.

		int[] xArray = new int[0]; //advanced feature I decided to show the size of the Integer array

		String[] sArray = {"StringArray"}; //advanced feature I decided to show the size of the String array


		IObjectProfileNode profile = ObjectProfiler.profile(s); //profile size method for the short primitive data type
        System.out.println("The Short data type size is " + profile.size() + " bytes and " + profile.size() * 8 + " bits");

        profile = ObjectProfiler.profile(i); //profile size method for the integer primitive data type
        System.out.println("The Integer data type is " + profile.size() + " bytes and " + profile.size() * 8 + " bits");

        profile = ObjectProfiler.profile(l); //profile size method for the long primitive data type
        System.out.println("The Long data type is " + profile.size() + " bytes and " + profile.size() * 8 + " bits");

		profile = ObjectProfiler.profile(f); //profile size method for the float primitive data type
        System.out.println("The Float data type is " + profile.size() + " bytes and " + profile.size() * 8 + " bits");

        profile = ObjectProfiler.profile(d); //profile size method for the double primitive data type
        System.out.println("The Double data type is " + profile.size() + " bytes and " + profile.size() * 8 + " bits");

        profile = ObjectProfiler.profile(b); //profile size method for the boolean primitive data type
        System.out.println("The Boolean data type is " + profile.size() + " bytes and " + profile.size() * 8 + " bits");

        profile = ObjectProfiler.profile(c); //profile size method for the char primitive data type
        System.out.println("The Char data type is " + profile.size() + " bytes and " + profile.size() * 8 + " bits");


        //This is the advanced feature I decided to do by showing the size of the interger array with 1 index
        profile = ObjectProfiler.profile(xArray);
        System.out.println("The size of the Integer array is " + profile.size() + " bytes and " + profile.size() * 8 + " bits");

        //This is another advanced feature I decided to do by showing the size of the String array that is declared
        profile = ObjectProfiler.profile(sArray);
        System.out.println("The size of the String array is " + profile.size() + " bytes and " + profile.size() * 8 + " bits");





	}
}

File: Lab/Lab6/usesizeof.c
#include <stdio.h>

int main(){


	short s;
	int i;
	long l;
	long long ll;
	long double ld;
	float f;
	double d;
	char c;
	char* c1;
	_Bool b;

	//this is all of the data types being used for this laboratory assignment


	printf("This is all in bytes \n");
	//this is all of the data types being represented in bytes by using the sizeof method which is avaialbe in the C programming language
	printf("Short: %d\n", sizeof(s));
	printf("Int: %d\n", sizeof(i));
	printf("Long: %d\n", sizeof(l));
	printf("Long Long: %d\n", sizeof(ll));
	printf("Long Double: %d\n", sizeof(ld));
	printf("Float: %d\n", sizeof(f));
	printf("Double: %d\n", sizeof(d));
	printf("Char %d\n", sizeof(c));
	printf("Char pointer: %d\n", sizeof(c1));
	printf("Boolean: %d\n", sizeof(b));

	printf("\n");

	printf("This is all in bits \n");
	//this is all of the data types being represented in bits
	printf("\n");


	/*
	in order the get the size of this variables in bits, I multiuple the sizeof method by 8 since the sizeof method computes
	the bytes and there are 8 bits in one byte. 
	*/
	printf("Short: %d\n", sizeof(s) * 8);
	printf("Int: %d\n", sizeof(i) * 8);
	printf("Long: %d\n", sizeof(l) * 8);
	printf("Long Long: %d\n", sizeof(ll) * 8);
	printf("Long Double: %d\n", sizeof(ld) * 8);
	printf("Float: %d\n", sizeof(f) * 8);
	printf("Double: %d\n", sizeof(d) * 8);
	printf("Char %d\n", sizeof(c) * 8);
	printf("Char pointer: %d\n", sizeof(c1) * 8);
	printf("Boolean: %d\n", sizeof(b) * 8);

	printf("\n");
/*
	this is the advanced features that I have implmented, I decided to implement an Array with 0 indexes, 1 index, and 5 indexes,
	and character of array with one element inside this array. 
	*/
	int iArray[0];

	int jArray[1];

	int xArray[5];

	char sArray = {"ArrayofCharacters"};

	//This printf statement will print out all of the data type size I have implemented in bytes and bits
	printf("Integer Array: %d bytes and %d bits \n", sizeof(iArray), sizeof(iArray) * 8);

	printf("Inter Array with 1 Array: %d bytes and %d bits \n", sizeof(jArray), sizeof(jArray) * 8);

	printf("Integer Array with 5 indexes: %d bytes and %d bits \n", sizeof(xArray), sizeof(xArray) * 8);

	printf("Character Array with one element: %d bytes and %d bits \n", sizeof(sArray), sizeof(sArray) * 8);





}

















